sandy/fy.py

```diff
@@ -445,24 +438,21 @@ def gls_update(self, zam, e, decay_data, xp, Vy=None, kind='cumulative',
 
         """
         data = self.data
-        S = _gls_setup(decay_data, kind, index_columns)
         # Filter FY data:
         fy_data = self.filter_by('ZAM', zam)\
                       .filter_by("E", e).data\
                       .set_index('ZAP')[['MT', 'FY', 'DFY']]
         # Divide the data:
         fy_type = fy_data.groupby('MT')
         IFY, CFY = [fy_type.get_group(x)['FY'] for x in fy_type.groups]
```

Comment:
here you are assuming that `for x in fy_type.groups` returns the MT numbers in increasing order.
I am not sure this is the case!
Maybe it just gives them back according to what MT number was found first.
I would avoid complexity by simplifying the format...something like

```
IFY = fy_data.query('MT==454').FY
CFY = fy_data.query('MT==459).FY
```

Reply:
Ah, PEP20 says that *sparse is better than dense*, then I would not do two operations on the same line